### DISCUSSION TITLE
NABU Internet Adapter testing

### DISCUSSION THREAD

**Discussion by sairuk** [retronas/discussions#29] [General]
Please comment with any testing you are able to do with this software

Documentation: https://github.com/danmons/retronas/wiki/NABU

**Comment by johntdavis84** [comment]
Hello,

I'd like to help with testing, but I'm not sure how.

I've been watching NABU videos on YouTube again, so I'm interested in trying out NABU emulation either through my MiSTer or MAME.

I'm aware of the modern [Nabu RetroNET](https://nabu.ca/) and [cloud CPM](https://forums.nabu.ca/viewtopic.php?t=8).

How does RetroNAS fit in? What does it provide that, say, MAME or the MiSTer can't do on their own? After looking at the docs, I'm a bit confused.

**Reply by sairuk** [reply]
I haven't looked into NABU in a long while. We implement the Internet Adapter software as a service which allows you to host the software library on the RetroNAS and or use the other functionality of the software with either a real or emulated machine.

Neither MAME nor MiSTer supported NABU when we added support for this iirc, once MAME supported the system you could in theory use a bitbanger device to connect to the software running on RetroNAS but MAME and MiSTer probably have their own implementations for handling software

I have a faint recollection of settng up MAME with a bitbanger to test the software but i can't find any notes on it atm.

**Reply by sairuk** [reply]
<del>NABU internet adapter doesn't appear to work correctly on trixie, runs but only displays a black screen.</del>

its fine, i just had net forwarding off at the time i was trying it out
